go/vt/topo/shard.go

```diff
@@ -578,3 +590,66 @@ func (ts *Server) GetTabletMapForShardByCell(ctx context.Context, keyspace, shar
 	}
 	return result, gerr
 }
+
+func shardFilePath(keyspace, shard string) string {
+	return path.Join(KeyspacesPath, keyspace, ShardsPath, shard, ShardFile)
+}
+
+// WatchShardData wraps the data we receive on the watch channel
+// The WatchShard API guarantees exactly one of Value or Err will be set.
+type WatchShardData struct {
+	Value *topodatapb.Shard
+	Err   error
+}
+
+// WatchShard will set a watch on the Shard object.
+// It has the same contract as conn.Watch, but it also unpacks the
+// contents into a Shard object
+func (ts *Server) WatchShard(ctx context.Context, keyspace, shard string) (*WatchShardData, <-chan *WatchShardData, CancelFunc) {
```

Comment:
We'll eventually need to harden this to make sure it stays connected to the topo.